docs/question/server.md

```diff
@@ -405,3 +407,17 @@ We recommend using the [`pl-figure`](../elements.md#pl-figure-element) and [`pl-
     <p>Here is a dynamically-rendered figure showing a line of slope $a = {{params.a}}$:</p>
     <img src="{{options.client_files_question_dynamic_url}}/fig.png" />
     ```
+
+## Testing questions with `test()`
+
+The question testing functionality is available under the "Settings" tab for a question.
+
+![Test buttons](test-buttons.png)
+
+The `test()` function is called to test the question. This function can be used to ensure that your question is working correctly. The `raw_submitted_answers` generated by the `test()` function will be used in place of the student's submission, and the remaining fields will be compared to the result of parsing and grading. If your code crashes, or if any of the fields are different, the test will report an issue.
+
+The `test()` function receives the output of `prepare()`, along with a `test_type` parameter. The `test_type` is either `correct`, `incorrect`, or `invalid`. Your function should generate `raw_submitted_answers` based on the inputs (e.g. `data["correct_answers"]`) and `test_type`. It should also set `score` and `feedback` for each answer.
+
+Most elements implement the `test()` function, and generate a `raw_submitted_answers` value based on `correct_answers` and `test_type`. If the element is unable to determine the needed inputs (i.e. `correct_answers` isn't set in `generate` or `prepare`), or the element doesn't implement the `test()` function, then you should implement the `test()` function at the question level in `server.py` to generate the needed inputs.
```

Comment:
Nit: can we somehow move this earlier in the description here? The vast majority of questions should be testable without `test()` in `server.py`; I'd like to better emphasize that fact.

Even if there's not a great way to do that, having this documented at all is an improvement 👍

Reply:
Changed in [3756d3b](https://github.com/PrairieLearn/PrairieLearn/pull/11910/commits/3756d3bddc05e259feb127eb3b078d49754ee9d6).